classes/core/PKPRequest.inc.php

```diff
@@ -421,7 +421,7 @@ function checkCSRF() {
 	 * Get the remote IP address of the current request.
 	 * @return string
 	 */
-	function getRemoteAddr() {
+	static function getRemoteAddr() {
```

Comment:
I know the kinds of warnings you're working to resolve with these, but longer term, our solution will be to have a `$request` instance available to call on (rather than static calls). So the warnings will be helpful in identifying these.

Reply:
On our side, we get an exception in this file when we try to upload a file somewhere in the application. Here is the exceptions I get for this event.

```
PHP Deprecated:  Non-static method PKPRequest::getUserVar() should not be called statically in C:\inetpub\wwwroot\ojs\lib\pkp\classes\form\Form.inc.php on line 369

PHP Deprecated:  Non-static method PKPRequest::_checkThis() should not be called statically in C:\inetpub\wwwroot\ojs\lib\pkp\classes\core\PKPRequest.inc.php on line 592

PHP Deprecated:  Methods with the same name as their class will not be constructors in a future version of PHP; Smarty_Compiler has a deprecated constructor in C:\inetpub\wwwroot\ojs\lib\pkp\lib\vendor\smarty\smarty\libs\Smarty_Compiler.class.php on line 35
```

Every changes we made is generally to prevent the application throwing an error and freezing at the same time at the user interface side. I know it's certainly not the right way to fix it, but we can't let the application freeze each time we try to upload a file. Can you hint us on how to resolve this problem?

This is also linked to the comment below.

Reply:
Yes, simply configuring PHP to direct warnings/errors to the log file and not the browser should resolve the UI freezing. We're working to resolve the warnings gradually but it'll take a few releases.